event/src/main/java/io/airlift/event/client/AbstractEventClient.java

```diff
@@ -50,6 +50,5 @@ public final <T> ListenableFuture<Void> post(EventGenerator<T> eventGenerator)
         return immediateFuture(null);
     }
 
-    protected abstract <T> void postEvent(T event)
-            throws IOException;
```

Comment:
The removal of exceptions on "production" classes that can be extended, needs to be thought about.  In this case, and code extending this that needs to throw an IOException would break.

Reply:
I've dropped this commit. Anything else that could break compatibility?